src/main/java/com/puppycrawl/tools/checkstyle/xpath/ElementNode.java

```diff
@@ -0,0 +1,97 @@
+package com.puppycrawl.tools.checkstyle.xpath;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import net.sf.saxon.type.Type;
+
+/**
+ * Represents element node of Xpath-tree.
+ *
+ * @author Timur Tibeyev
+ */
+public class ElementNode extends AbstractNode {
+    /** Represents value of TokenTypes#IDENT. */
+    private String image;
```

Comment:
'image` is not a correct name to represent a node type. PMD uses that name due to the fact that their API work with that name, and that is a historical problem of their design.